big-red-shuttle/ScheduleViewController.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  ScheduleViewController.swift
+//  
+//
+//  Created by Monica Ong on 10/30/16.
+//
+//
+
+import UIKit
+
+class ScheduleViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, ScheduleBarDelegate {
+    
+    private var table: UITableView!
+    private var stops: [String]!
+    private var times: [String]!
+    private var looptimes: [String]!
+    private var scheduleBar: ScheduleBar!
+    
+    private let identifier: String = "Schedule Cell"
+    private let cellHeight: CGFloat = 55
+    private let barHeight: CGFloat = 55
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        //Data
+        //Q: How are we differentiating bt Stewart and University & Stewart and University- up to Thurston?
+        stops = ["Jessup & Program House Drive", "Kelvin & Wyckoff", "Stewart & University", "Baker Flagpole", "Schwartz Center", "Dryden & Eddy", "Stewart & University 2", "Thurston & Cradit Farm", "Jessup & Program House Drive", "Kelvin & Wyckoff", "Stewart & University", "Baker Flagpole", "Schwartz Center", "Dryden & Eddy", "Stewart & University 2", "Thurston & Cradit Farm"]
+        times = ["12:00 am", "12:02 am", "12:04 am", "12:06 am", "12:07 am", "12:08 am", "12:10 am", "12:12 am", "12:14 am", "12:16 am", "12:18 am", "12:20 am", "12:22 am", "12:24 am", "12:26 am", "12:28 am"]
+        looptimes = ["12:00 am", "12:07 am", "12:12 am", "12:16 am", "12:18 am", "12:28 am"]
+        
+        //Bar
+        scheduleBar = ScheduleBar(frame: CGRect(x: 0, y: 0, width: self.view.bounds.width, height: barHeight))
+        scheduleBar.setUp(buttonsData: looptimes, selected: 2)
+        scheduleBar.delegateSB = self
+        self.view.addSubview(scheduleBar)
+        
+        //Table
+        table = UITableView(frame: CGRect(x: 0, y: scheduleBar.frame.maxY, width: view.bounds.width, height: view.bounds.height - (navigationController?.navigationBar.bounds.height)! - (navigationController?.tabBarController?.tabBar.bounds.height)! - scheduleBar.bounds.height))
+        table.separatorStyle = UITableViewCellSeparatorStyle.none
+        table.delegate = self
+        table.dataSource = self
+        self.view.addSubview(table)
+        
+        //Nav
+        self.navigationController?.navigationBar.topItem?.title = "Schedule"
+        
+    }
+    
+    override func viewWillAppear(_ animated: Bool){
+        super.viewWillAppear(animated)
+        table.register(ScheduleTableViewCell.classForCoder(), forCellReuseIdentifier: identifier)
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    
+    // MARK: - Schedule Bar delegate
+    
+    func selectCell(button: UIButton) {
+        let time = looptimes[button.tag]
+        times.index(of: time)
```

Comment:
Just as an aside, this would also be a horrible way to implement something like this in the real world because of localization and view/model separation. If you add localization for any language, this mechanism would not work. Technically all UI facing strings should never determine code execution because they should generally all be created using `NSLocalizedString()` (but I can't enforce all the rules lol). I'd have to look at the actual project and not just the code to know what's going on here, but just throwing in my 2 cents.